web/src/narrow_history.ts

```diff
@@ -30,11 +30,14 @@ function _save_narrow_state(): void {
         return;
     }
     const narrow_offset = $narrow_row.get_offset_to_window().top;
-    const narrow_data = {
+    const state_data = {
         narrow_pointer,
         narrow_offset,
     };
-    history.replaceState(narrow_data, "", window.location.href);
+    if (history.state) {
+        Object.assign(state_data, history.state);
+    }
+    history.replaceState(state_data, "", window.location.href);
```

Comment:
This `object.assign` logic is a bit confusing; what are the intended semantics in terms of which fields set on `state_data` a few lines above this should overwrite from `history.state`? I wonder if this could be written more clearly.

Reply:
Simplified this logic. 

BTW `Object.assign(state_data, history.state);` was wrong earlier.